pkg/dynamiclistener/server.go

```diff
@@ -111,6 +116,9 @@ func NewServer(ctx context.Context, listenConfigStorage ListenConfigStorage,
 		httpPort:            httpPort,
 		httpsPort:           httpsPort,
 		certs:               map[string]*tls.Certificate{},
+		// while starting server, the default value of setting is not set yet, hence starting with a hardcoded value,
+		// the setting cert-expiration-check-interval can be updated later to change this
+		certCheckIntervalInDays: 7,
```

Comment:
You should just catch it on demand. Introduce the method where this param gets defaulted to 7 days if setting doesn't exist. Think of renaming the parameter.